web/react/components/edit_post_modal.jsx

```diff
@@ -26,79 +34,102 @@ module.exports = React.createClass({
+    }
+    componentDidMount() {
         var self = this;
 
-        $(this.refs.modal.getDOMNode()).on('hidden.bs.modal', function(e) {
-            self.setState({editText: "", title: "", channel_id: "", post_id: "", comments: 0, refocusId: "", error: ''});
+        $(React.findDOMNode(this.refs.modal)).on('hidden.bs.modal', function onHidden() {
+            self.setState({editText: '', title: '', channel_id: '', post_id: '', comments: 0, refocusId: '', error: ''});
         });
 
-        $(this.refs.modal.getDOMNode()).on('show.bs.modal', function(e) {
+        $(React.findDOMNode(this.refs.modal)).on('show.bs.modal', function onShow(e) {
             var button = e.relatedTarget;
-            self.setState({ editText: $(button).attr('data-message'), title: $(button).attr('data-title'), channel_id: $(button).attr('data-channelid'), post_id: $(button).attr('data-postid'), comments: $(button).attr('data-comments'), refocusId: $(button).attr('data-refoucsid') });
+            self.setState({editText: $(button).attr('data-message'), title: $(button).attr('data-title'), channel_id: $(button).attr('data-channelid'), post_id: $(button).attr('data-postid'), comments: $(button).attr('data-comments'), refocusId: $(button).attr('data-refoucsid')});
         });
 
-        $(this.refs.modal.getDOMNode()).on('shown.bs.modal', function(e) {
+        $(React.findDOMNode(this.refs.modal)).on('shown.bs.modal', function onShown() {
             self.refs.editbox.resize();
         });
-    },
-    getInitialState: function() {
-        return { editText: "", title: "", post_id: "", channel_id: "", comments: 0, refocusId: "" };
-    },
-    render: function() {
-        var error = this.state.error ? <div className='form-group has-error'><br /><label className='control-label'>{ this.state.error }</label></div> : <div className='form-group'><br /></div>;
+    }
+    render() {
+        var error = (<div className='form-group'><br /></div>);
+        if (this.state.error) {
+            error = (<div className='form-group has-error'><br /><label className='control-label'>{this.state.error}</label></div>);
+        }
 
         return (
-            <div className="modal fade edit-modal" ref="modal" id="edit_post" role="dialog" tabIndex="-1" aria-hidden="true">
-              <div className="modal-dialog modal-push-down">
-                <div className="modal-content">
-                  <div className="modal-header">
-                    <button type="button" className="close" data-dismiss="modal" aria-label="Close" onClick={this.handleEditClose}><span aria-hidden="true">&times;</span></button>
-                    <h4 className="modal-title">Edit {this.state.title}</h4>
+            <div
+                className='modal fade edit-modal'
+                ref='modal'
+                id='edit_post'
+                role='dialog'
+                tabIndex='-1'
+                aria-hidden='true' >
+              <div className='modal-dialog modal-push-down'>
```

Comment:
I noticed when going through my own files that there's some render methods that are indented with 2 spaces instead of 4. This is probably a good time to fix that.